query-connector/src/auth.ts

```diff
@@ -2,5 +2,57 @@ import NextAuth from "next-auth";
 import Keycloak from "next-auth/providers/keycloak";
 
 export const { handlers, signIn, signOut, auth } = NextAuth({
-  providers: [Keycloak],
+  providers: [
+    Keycloak({
+      clientId: process.env.AUTH_KEYCLOAK_ID,
+      clientSecret: process.env.AUTH_KEYCLOAK_SECRET,
+      issuer: process.env.AUTH_KEYCLOAK_ISSUER,
+    }),
+  ],
+  callbacks: {
+    /**
+     * JWT callback to store Keycloak user data in the token.
+     * @param token The root object containing the JWT properties.
+     * @param token.token The current JWT token.
+     * @param token.profile The user profile returned from Keycloak.
+     * @returns The updated JWT token with user details.
+     */
+    async jwt({ token, profile }) {
+      console.log("JWT Callback - Initial Token:", token);
+      console.log("JWT Callback - Profile:", profile);
+
+      if (profile) {
+        token.id = profile.sub;
+        token.username = profile.preferred_username || profile.email;
+        token.email = profile.email;
+        token.firstName = profile.given_name;
+        token.lastName = profile.family_name;
+      }
+
+      console.log("JWT Callback - Final Token:", token);
+      return token;
+    },
+    /**
+     * Session callback to pass user data to the session object.
+     * @param session The root object containing session properties.
+     * @param session.session The session object.
+     * @param session.token The JWT token containing user data.
+     * @returns The updated session object with user details.
+     */
+    async session({ session, token }) {
+      console.log("Session Callback - Initial Session:", session);
+      console.log("Session Callback - Token:", token);
+
+      session.user = {
+        ...token,
+        id: typeof token.id === "string" ? token.id : "",
+        email: token.email || "",
```

Comment:
Nice, glad we're only passing these fields and not the names to the client side.